### PR TITLE
Cleaned up webhook tester for Python 3 and improved error handling

### PR DESCRIPTION
**Pull request by Pontiky**
## Description

- Each time we use `webhook_tester.py` and we forget to start a PA server, the script is crashing because it can't send the webhook to the provided URL. I tweaked it to signal there was an error sending the webhook and allowed to send it again if the PA server wasn't started instead of crashing.
 - The script doesn't exit anymore if we don't choose the webhook type in the first question. It asks again to choose a type.
 - The last changes are only about code readability. I made a lot of standardization between the different questions and webhook types. As PA supports only Python 3.7+, I used the new syntaxes provided to improve code readability of print functions.

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (would cause existing functionality to change)

## Motivation and Context

Each time I use `webhook_tester.py` and I forget to start a PA server, the script is crashing and that's a pain to restart and enter all the webhook informations again, especially when we have to set encounter informations.
The code was also quite difficult to read, so I cleaned it up.

## How Has This Been Tested?

Tested each type of webhook. New errors are handled correctly.
I didn't changed or added any functionalities, just error handling and code cleanup.

## Wiki Update

- [ ] This change requires an update to the Wiki.
- [x] This change does not require an update to the Wiki.

### CODE REVIEW

**Comment by Pontiky** [comment]
### Comparison tests 
<details><summary>Test 1: PA is not started (dev branch)</summary>

```shell
(pokealarm_env) anthony@anthony-K54LY:~/local_pokealarm/pokealarm$ python tools/webhook_test.py 
What kind of webhook would you like to send?(put in a number)

1: pokemon
  2: pokestop
  3: gym
  4: egg
  5: raid
  6: weather
  7: quest
  8: invasion
> 1
What is the URL of where you would like to send the webhook? (default: http://127.0.0.1:4000)
> 
Assuming http://127.0.0.1:4000 as webhook URL
Does location matter or do you use geofences? (Y/N)
> 
Enter Pokemon ID
> 
Not a valid number. Defaulting to 149
Gender (1-3)
> 
Not a valid number. Defaulting to 3
Which form of Dragonite would you like? (default: Normal)
Normal, Shadow, Purified, Costume 2020
> 
Not a valid value, using default
Encounters enabled?
> 
What type of weather? (number only)(Default: None)

  000: None
  001: Clear
  002: Rain
  003: Partly Cloudy
  004: Cloudy
  005: Windy
  006: Snow
  007: Fog

> 
Not a valid number. Defaulting to None
Is this mon boosted by the weather? (y/n)
> 
Traceback (most recent call last):
  File "/home/anthony/pokealarm_env/lib/python3.8/site-packages/urllib3/connection.py", line 159, in _new_conn
    conn = connection.create_connection(
  File "/home/anthony/pokealarm_env/lib/python3.8/site-packages/urllib3/util/connection.py", line 84, in create_connection
    raise err
  File "/home/anthony/pokealarm_env/lib/python3.8/site-packages/urllib3/util/connection.py", line 74, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/anthony/pokealarm_env/lib/python3.8/site-packages/urllib3/connectionpool.py", line 670, in urlopen
    httplib_response = self._make_request(
  File "/home/anthony/pokealarm_env/lib/python3.8/site-packages/urllib3/connectionpool.py", line 392, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/lib/python3.8/http/client.py", line 1256, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib/python3.8/http/client.py", line 1302, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.8/http/client.py", line 1251, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.8/http/client.py", line 1011, in _send_output
    self.send(msg)
  File "/usr/lib/python3.8/http/client.py", line 951, in send
    self.connect()
  File "/home/anthony/pokealarm_env/lib/python3.8/site-packages/urllib3/connection.py", line 187, in connect
    conn = self._new_conn()
  File "/home/anthony/pokealarm_env/lib/python3.8/site-packages/urllib3/connection.py", line 171, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f96a1f469d0>: Failed to establish a new connection: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/anthony/pokealarm_env/lib/python3.8/site-packages/requests/adapters.py", line 439, in send
    resp = conn.urlopen(
  File "/home/anthony/pokealarm_env/lib/python3.8/site-packages/urllib3/connectionpool.py", line 726, in urlopen
    retries = retries.increment(
  File "/home/anthony/pokealarm_env/lib/python3.8/site-packages/urllib3/util/retry.py", line 446, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='127.0.0.1', port=4000): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f96a1f469d0>: Failed to establish a new connection: [Errno 111] Connection refused'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "tools/webhook_test.py", line 537, in <module>
    resp = requests.post(url, json=payload, timeout=5)
  File "/home/anthony/pokealarm_env/lib/python3.8/site-packages/requests/api.py", line 116, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/home/anthony/pokealarm_env/lib/python3.8/site-packages/requests/api.py", line 60, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/anthony/pokealarm_env/lib/python3.8/site-packages/requests/sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/anthony/pokealarm_env/lib/python3.8/site-packages/requests/sessions.py", line 646, in send
    r = adapter.send(request, **kwargs)
  File "/home/anthony/pokealarm_env/lib/python3.8/site-packages/requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=4000): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f96a1f469d0>: Failed to establish a new connection: [Errno 111] Connection refused'))
(pokealarm_env) anthony@anthony-K54LY:~/local_pokealarm/pokealarm$ 
```
</details>

<details><summary>Test 1: PA is not started (PR808)</summary>

```shell
(pokealarm_env) anthony@anthony-K54LY:~/local_pokealarm/pokealarm$ python tools/webhook_test.py 
What kind of webhook would you like to send? (1-8)

  1: pokemon
  2: pokestop
  3: gym
  4: egg
  5: raid
  6: weather
  7: quest
  8: invasion

> 1 
What is the URL of where you would like to send the webhook? (default: http://127.0.0.1:4000)
> 
Using default: http://127.0.0.1:4000
Does location matter or do you use geofences? (y/n)
> 
Pokemon ID (default: 149)
> 
Invalid number, using default: 149
Gender (1-3) (default: 3)
> 
Invalid number, using default: 3
Which form of Dragonite would you like? (default: Normal)
Normal, Shadow, Purified, Costume 2020
> 
Invalid value, using default
Encounter enabled? (y/n)
> 
Weather type (default: 0)

  000: None
  001: Clear
  002: Rain
  003: Partly Cloudy
  004: Cloudy
  005: Windy
  006: Snow
  007: Fog

> 
Invalid number, using default: None
Pokemon weather boosted? (y/n)
> 
Cannot send webhook to http://127.0.0.1:4000. Check the webhook URL or if your PA server is running.
Send again? (y/n)
> y
Cannot send webhook to http://127.0.0.1:4000. Check the webhook URL or if your PA server is running.
Send again? (y/n)
> y
Cannot send webhook to http://127.0.0.1:4000. Check the webhook URL or if your PA server is running.
Send again? (y/n)
> y
Cannot send webhook to http://127.0.0.1:4000. Check the webhook URL or if your PA server is running.
Send again? (y/n)
> y
Notification successful. Returned code 200
Send again? (y/n)
> 
```
</details>

<details><summary>Test 2: Wrong webhook type selected (dev branch)</summary>

```shell
(pokealarm_env) anthony@anthony-K54LY:~/local_pokealarm/pokealarm$ python tools/webhook_test.py 
What kind of webhook would you like to send?(put in a number)

  1: pokemon
  2: pokestop
  3: gym
  4: egg
  5: raid
  6: weather
  7: quest
  8: invasion
> 
Must put in valid webhook type
(pokealarm_env) anthony@anthony-K54LY:~/local_pokealarm/pokealarm$ 
```
</details>

<details><summary>Test 2: Wrong webhook type selected (PR808)</summary>

```shell
(pokealarm_env) anthony@anthony-K54LY:~/local_pokealarm/pokealarm$ python tools/webhook_test.py 
What kind of webhook would you like to send? (1-8)

  1: pokemon
  2: pokestop
  3: gym
  4: egg
  5: raid
  6: weather
  7: quest
  8: invasion

> 
Invalid webhook type. Please try again..
> 
Invalid webhook type. Please try again..
> 
Invalid webhook type. Please try again..
> 875
Invalid webhook type. Please try again..
> 9
Invalid webhook type. Please try again..
> 1
What is the URL of where you would like to send the webhook? (default: http://127.0.0.1:4000)
> 
...
```
</details>

### Description

With `Test 1`, you can also see there are some changes about the syntax. I added the `default` information where they were missing and standardized some cases.

- Same informations are asked as the same way through the differents types (`Pokemon ID` everywhere, not `Enter pokemon id for raid (5: raid)` then `Pokemon ID? (7: quest)` or `Enter Pokemon ID (1: pokemon)` )
- With the 1st point, all questions which need a yes/no as answer will show `(y/n)`
- A lot of missing default values have been added
- Removed or added a lot of missing/extra spaces and useless `\n` in `print()` functions in Python
- Improved code readability of `print()` by using the `formatted string literal` introduced in Python 3.6.

<details><summary>Example</summary>

```python
# dev branch
print("Which form of " + data["pokemon"][monster_id_formatted] + ' would you like? (default: ' + data['forms'][monster_id_formatted][default_form_id] + ')\n' + forms_formatted + '\n>', end=' ')
```
```python
# PR808
print(f"Which form of {data['pokemon'][monster_id_formatted]} would you like? (default: {data['forms'][monster_id_formatted][default_form_id]})")
print(forms_formatted, end='\n> ')
```
</details>

**Comment by CarlosVMX1** [comment]
Tested this PR with different options and works good, then I use it with PA off and it asks to check the webhook address with an option to retry, so all works as expected.

File: tools/webhook_test.py [hunk 1]
```diff
@@ -377,177 +375,186 @@ def monster_form(webhook_field, monster_id):
...
         lat, lng = map(float, coordinates.split(","))
     payload["message"]["latitude"] = lat
     payload["message"]["longitude"] = lng
 
 if type == whtypes["1"]:
-    print("Enter Pokemon ID\n>", end=' ')
-    int_or_default("pokemon_id")
-    print("Gender (1-3)\n>", end=' ')
-    int_or_default("gender")
+    print("Pokemon ID (default: 149)", end='\n> ')
```
Comment (billyjbryant) [inline tools/webhook_test.py:428]: I see we have some discrepancies. Can you change this to `Monster`?

File: tools/webhook_test.py [hunk 2]
```diff
@@ -377,177 +375,186 @@ def monster_form(webhook_field, monster_id):
...
+            print("Count towards big Magikarp medal? (y/n)", end='\n> ')
             if input() in truthy:
                 payload["message"]["weight"] = 14.0
-        print("Monster level?\n>", end=' ')
-        int_or_default("pokemon_level")
-    print("What type of weather? (number only)(Default: None)\n"
-          + weather_formatted + "\n>", end=' ')
-    int_or_default("weather")
-    print("Is this mon boosted by the weather? (y/n)\n>", end=' ')
+        print("Pokemon level", end='\n> ')
```
Comment (billyjbryant) [inline tools/webhook_test.py:457]: Same here

File: tools/webhook_test.py [hunk 3]
```diff
@@ -377,177 +375,186 @@ def monster_form(webhook_field, monster_id):
...
+    print(teams_formatted, end='\n> ')
     get_and_validate_team()
 elif type == whtypes["5"]:
     gym_cache()
-    print("Enter pokemon id for raid\n>", end=' ')
-    int_or_default("pokemon_id")
-    print("Enter form id for raid monster\n>", end=' ')
-    int_or_default("form")
-    print("Which team?(put in a number)\n" + teams_formatted + "\n>", end=' ')
+    print("Pokemon ID (default: 150)", end='\n> ')
```
Comment (billyjbryant) [inline tools/webhook_test.py:482]: Here as well. We are not too consistent it seems :D

**Review by billyjbryant — APPROVED** [review]

**Pull request merged**